sphinxgallery/gen_rst.py

```diff
@@ -118,133 +79,134 @@ def flush(self):
-        unsorted[count][1] = total_lines - docstr_lines
-        unsorted[count][0] = exmpl
-    index = np.lexsort((unsorted[:, 0].astype(np.str),
-                        unsorted[:, 1].astype(np.float)))
-    if not len(unsorted):
-        return []
-    return np.array(unsorted[index][:, 0]).tolist()
-
-
-def generate_dir_rst(src_dir, target_dir, gallery_conf,
-                     plot_gallery, seen_backrefs):
-    """Generate the rst file for an example directory"""
-    if not os.path.exists(os.path.join(src_dir, 'README.txt')):
-        print(80 * '_')
-        print('Example directory %s does not have a README.txt file' %
-              src_dir)
-        print('Skipping this directory')
-        print(80 * '_')
-        return ""  # because string is an expected return type
+        elif split_code_block.startswith('#'):
+            sub_pat = re.compile('^#', flags=re.M)
+            coment_block = dedent(re.sub(sub_pat, '', split_code_block))
+            blocks.append(('text', coment_block))
+        else:
+            blocks.append(('code', split_code_block))
 
-    fhindex = open(os.path.join(src_dir, 'README.txt')).read()
-    if not os.path.exists(target_dir):
-        os.makedirs(target_dir)
-    sorted_listdir = line_count_sort(os.listdir(src_dir),
-                                     src_dir)
-    for fname in sorted_listdir:
-        if fname.endswith('py'):
-            generate_file_rst(fname, target_dir, src_dir, plot_gallery)
-            new_fname = os.path.join(src_dir, fname)
-            _, snippet, _ = extract_docstring(new_fname, True)
-            write_backreferences(seen_backrefs, gallery_conf,
-                               target_dir, fname, snippet)
+    return blocks
 
-            fhindex += _thumbnail_div(target_dir, fname, snippet)
-            fhindex += """
 
-.. toctree::
-   :hidden:
+def codestr2rst(codestr):
+    """Return reStructuredText code block from code string"""
+    code_directive = "\n.. code-block:: python\n\n"
+    indented_block = '    ' + codestr.replace('\n', '\n    ')
+    return code_directive + indented_block
 
-   /%s/%s\n""" % (target_dir, fname[:-3])
 
+def text2string(content):
+    """Returns a string without the extra triple quotes"""
+    try:
+        return ast.literal_eval(content)+'\n'
+    except:
+        return content
 
-# clear at the end of the section
-    fhindex += """.. raw:: html\n
-    <div style='clear:both'></div>\n\n"""
 
-    return fhindex
+def extract_intro(filename):
+    """ Extract the first paragraph of module-level docstring. max:95 char"""
+
+    first_text = split_code_and_text_blocks(filename)[0][-1]
+
+    paragraphs = first_text.split('\n\n')
+    if len(first_text) > 1:
+        first_par = re.sub('\n', ' ', paragraphs[1])
+        first_par = ((first_par[:95] + '...')
+                     if len(first_par) > 95 else first_par)
+    else:
+        raise ValueError("Missing first paragraph."
```

Comment:
You might want a slightly better error message here. For example mentioning that you need a title (as a rst header) and a short explanation paragraph or something better.